### DISCUSSION TITLE
Multiple wifi networks for roaming

### DISCUSSION THREAD

**Discussion by mntn-xyz** [meshtastic/firmware#5593] [Ideas]
I have a T-Deck and bring it with me to multiple locations. Each location has a different network with a unique SSID. It would be helpful if the device could store multiple network configurations and automatically join the strongest network.

**Comment by fifieldt** [comment]
Not a terrible idea, but we're very limited in storage space on these devices. Sometimes we remove code to save a few bytes, and wifi names and passwords are fairly long.

**Comment by mntn-xyz** [comment]
Yes, space is definitely an issue. This might be best as a module that is only built for devices with large storage or SD card support and adequate PSRAM. Probably not an urgent priority, but it would be nice to have better support for standalone devices especially as the new device-ui matures.

**Reply by madeofstown** [reply]
This is also something that could be achieved within the apps. App could automate sending current active WiFi network info to the node.

**Reply by mntn-xyz** [reply]
That's also a good idea, but it doesn't help with standalone devices not paired with a phone. I don't carry a smartphone so it wouldn't work for my use case.

**Reply by garthvh** [reply]
It is not about overall storage space, it is about the max packet size of config sections in order to support lora administration functionality, both the network config and MQTT are already very close to the limit and can't support adding another long username and password field.

**Reply by mntn-xyz** [reply]
That's one reason I mentioned SD cards. I imagine that map tiles for instance will need to be loaded separately, not through LoRa admin. A `wifi.ini` or something on the SD card would allow for fairly complex roaming wifi setups. It requires manual provisioning but that is better than nothing.

**Reply by garthvh** [reply]
You can only use one connection at a time in meshtastic, the MUI uses the connection so WiFi won't be available.

**Reply by mntn-xyz** [reply]
Thanks, I had no idea the new UI used the connection - this would definitely be a problem. I was hoping the device could seamlessly bridge LoRa and wifi meshes by itself. Looks like this won't be possible unless the connection model changes in the future.
